src/Components/Web.JS/src/Services/NavigationEnhancement.ts

```diff
@@ -217,10 +221,23 @@ export async function performEnhancedPageLoad(internalDestinationHref: string, f
         return;
       }
 
-      if (!response.redirected && !isGetRequest && response.url !== location.href && isSuccessResponse) {
-        isNonRedirectedPostToADifferentUrlMessage = `Cannot perform enhanced form submission that changes the URL (except via a redirection), because then back/forward would not work. Either remove this form\'s \'action\' attribute, or change its method to \'get\', or do not mark it as enhanced.\nOld URL: ${location.href}\nNew URL: ${response.url}`;
+      if (!response.redirected && !isGetRequest && isSuccessResponse) {
+        // If this is the result of a form post that didn't trigger a redirection.
+        if (!isForSamePath(response)) {
+          // In this case we don't want to push the currentContentUrl to the history stack because we don't know if this is a location
+          // we can navigate back to (as we don't know if the location supports GET) and we are not able to replicate the Resubmit form?
+          // browser behavior.
+          // The only case where this is acceptable is when the last content URL, is the same as the URL for the form we posted to.
+          isNonRedirectedPostToADifferentUrlMessage = `Cannot perform enhanced form submission that changes the URL (except via a redirection), because then back/forward would not work. Either remove this form\'s \'action\' attribute, or change its method to \'get\', or do not mark it as enhanced.\nOld URL: ${location.href}\nNew URL: ${response.url}`;
+        } else {
+          // The url on the browser might be out of data, so push an entry to the stack to update the url in place.
+          history.pushState(null, '', currentContentUrl);
```

Comment:
I think you'll have to do that conditionally (i.e., only if `location.href !== currentContentUrl`) otherwise it will keep appending duplicate URLs to the history stack.

Note that I checked and `history.pushState(null, '', location.href)` is *not* a no-op. It really does append duplicates, making 'back' hard to use.

Reply:
@SteveSandersonMS you are right.

I'm not able to run the tests from VS, so I am trying to work this out on the CI and will update the PR when its ready. I still need to make sure the back button works.